shared/src/commonMain/kotlin/org/hyperskill/app/step/presentation/StepActionDispatcher.kt

```diff
@@ -26,16 +45,70 @@ class StepActionDispatcher(
                     .fold(
                         onSuccess = {
                             sentryInteractor.finishTransaction(sentryTransaction)
-                            onNewMessage(Message.StepLoaded.Success(it))
+                            onNewMessage(
+                                Message.StepLoaded.Success(
+                                    step = it,
+                                    practiceStatus = if (action.stepRoute is StepRoute.Learn) {
+                                        PracticeStatus.AVAILABLE
+                                    } else {
+                                        PracticeStatus.UNAVAILABLE
+                                    }
```

Comment:
Why this logic is in Dispatcher? In my opinion it should be placed in Reducer, because it is static and it is not rely on some data source or other stuff.